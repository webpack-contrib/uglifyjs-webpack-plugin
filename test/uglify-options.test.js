import UglifyJsPlugin from '../src/index';
import {
  cleanErrorStack,
  createCompiler,
  compile,
} from './helpers';

describe('when applied with uglify-es options', () => {
  it('matches snapshot for `toplevel` option', () => {
    const compiler = createCompiler({
      entry: `${__dirname}/fixtures/import-export/entry.js`,
      output: {
        path: `${__dirname}/dist-import-export`,
        filename: '[name].js',
        chunkFilename: '[id].[name].js',
      },
    });

    new UglifyJsPlugin({
      uglifyOptions: {
        toplevel: true,
        mangle: false,
        warnings: true,
        output: {
          beautify: true,
        },
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('import-export: errors');
      expect(warnings).toMatchSnapshot('import-export: warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(`import-export: ${file}`);
        }
      }
    });
  });

  it('matches snapshot for `nameCache` option', () => {
    const compiler = createCompiler({
      entry: `${__dirname}/fixtures/import-export/entry.js`,
      output: {
        path: `${__dirname}/dist-import-export`,
        filename: '[name].js',
        chunkFilename: '[id].[name].js',
      },
    });

    new UglifyJsPlugin({
      uglifyOptions: {
        nameCache: {},
        mangle: false,
        warnings: true,
        output: {
          beautify: true,
        },
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('import-export: errors');
      expect(warnings).toMatchSnapshot('import-export: warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(`import-export: ${file}`);
        }
      }
    });
  });

  it('matches snapshot for `keep_classnames` option', () => {
    const compiler = createCompiler({
      entry: `${__dirname}/fixtures/import-export/entry.js`,
      output: {
        path: `${__dirname}/dist-import-export`,
        filename: '[name].js',
        chunkFilename: '[id].[name].js',
      },
    });

    new UglifyJsPlugin({
      uglifyOptions: {
        keep_classnames: true,
        mangle: false,
        warnings: true,
        output: {
          beautify: true,
        },
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('import-export: errors');
      expect(warnings).toMatchSnapshot('import-export: warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(`import-export: ${file}`);
        }
      }
    });
  });

  it('matches snapshot for `keep_fnames` option', () => {
    const compiler = createCompiler({
      entry: `${__dirname}/fixtures/import-export/entry.js`,
      output: {
        path: `${__dirname}/dist-import-export`,
        filename: '[name].js',
        chunkFilename: '[id].[name].js',
      },
    });

    new UglifyJsPlugin({
      uglifyOptions: {
        keep_fnames: true,
        mangle: false,
        warnings: true,
        output: {
          beautify: true,
        },
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('import-export: errors');
      expect(warnings).toMatchSnapshot('import-export: warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(`import-export: ${file}`);
        }
      }
    });
  });

  it('matches snapshot for `safari10` option', () => {
    const compiler = createCompiler({
      entry: `${__dirname}/fixtures/import-export/entry.js`,
      output: {
        path: `${__dirname}/dist-import-export`,
        filename: '[name].js',
        chunkFilename: '[id].[name].js',
      },
    });

    new UglifyJsPlugin({
      uglifyOptions: {
        safari10: true,
        mangle: false,
        warnings: true,
        output: {
          beautify: true,
        },
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('import-export: errors');
      expect(warnings).toMatchSnapshot('import-export: warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(`import-export: ${file}`);
        }
      }
    });
  });
});
