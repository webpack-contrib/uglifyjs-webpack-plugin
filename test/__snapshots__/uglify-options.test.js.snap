// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with uglify-es options matches snapshot for \`keep_classnames\` option: errors 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`keep_classnames\` option: main.js 1`] = `
"webpackJsonp([ 0 ], [ function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    Object.defineProperty(__webpack_exports__, \\"__esModule\\", {
        value: !0
    });
    var __WEBPACK_IMPORTED_MODULE_0__dep__ = __webpack_require__(1);
    __webpack_exports__.default = function() {
        const b = __WEBPACK_IMPORTED_MODULE_0__dep__.b, baz = \`baz\${Math.random()}\`;
        return () => ({
            a: b + __WEBPACK_IMPORTED_MODULE_0__dep__.a + baz,
            b: b,
            baz: baz
        });
    };
}, function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    __webpack_exports__.a = \\"bar\\", __webpack_exports__.b = \\"foo\\";
} ], [ 0 ]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`keep_classnames\` option: manifest.js 1`] = `
"!function(modules) {
    var parentJsonpFunction = window.webpackJsonp;
    window.webpackJsonp = function(chunkIds, moreModules, executeModules) {
        for (var moduleId, chunkId, result, i = 0, resolves = []; i < chunkIds.length; i++) chunkId = chunkIds[i], 
        installedChunks[chunkId] && resolves.push(installedChunks[chunkId][0]), installedChunks[chunkId] = 0;
        for (moduleId in moreModules) Object.prototype.hasOwnProperty.call(moreModules, moduleId) && (modules[moduleId] = moreModules[moduleId]);
        for (parentJsonpFunction && parentJsonpFunction(chunkIds, moreModules, executeModules); resolves.length; ) resolves.shift()();
        if (executeModules) for (i = 0; i < executeModules.length; i++) result = __webpack_require__(__webpack_require__.s = executeModules[i]);
        return result;
    };
    var installedModules = {}, installedChunks = {
        1: 0
    };
    function __webpack_require__(moduleId) {
        if (installedModules[moduleId]) return installedModules[moduleId].exports;
        var module = installedModules[moduleId] = {
            i: moduleId,
            l: !1,
            exports: {}
        };
        return modules[moduleId].call(module.exports, module, module.exports, __webpack_require__), 
        module.l = !0, module.exports;
    }
    __webpack_require__.m = modules, __webpack_require__.c = installedModules, __webpack_require__.d = function(exports, name, getter) {
        __webpack_require__.o(exports, name) || Object.defineProperty(exports, name, {
            configurable: !1,
            enumerable: !0,
            get: getter
        });
    }, __webpack_require__.n = function(module) {
        var getter = module && module.__esModule ? function() {
            return module.default;
        } : function() {
            return module;
        };
        return __webpack_require__.d(getter, \\"a\\", getter), getter;
    }, __webpack_require__.o = function(object, property) {
        return Object.prototype.hasOwnProperty.call(object, property);
    }, __webpack_require__.p = \\"\\", __webpack_require__.oe = function(err) {
        throw console.error(err), err;
    };
}([]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`keep_classnames\` option: warnings 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`keep_fnames\` option: errors 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`keep_fnames\` option: main.js 1`] = `
"webpackJsonp([ 0 ], [ function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    Object.defineProperty(__webpack_exports__, \\"__esModule\\", {
        value: !0
    });
    var __WEBPACK_IMPORTED_MODULE_0__dep__ = __webpack_require__(1);
    __webpack_exports__.default = function Foo() {
        const b = __WEBPACK_IMPORTED_MODULE_0__dep__.b, baz = \`baz\${Math.random()}\`;
        return () => ({
            a: b + __WEBPACK_IMPORTED_MODULE_0__dep__.a + baz,
            b: b,
            baz: baz
        });
    };
}, function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    __webpack_exports__.a = \\"bar\\", __webpack_exports__.b = \\"foo\\";
} ], [ 0 ]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`keep_fnames\` option: manifest.js 1`] = `
"!function(modules) {
    var parentJsonpFunction = window.webpackJsonp;
    window.webpackJsonp = function webpackJsonpCallback(chunkIds, moreModules, executeModules) {
        for (var moduleId, chunkId, result, i = 0, resolves = []; i < chunkIds.length; i++) chunkId = chunkIds[i], 
        installedChunks[chunkId] && resolves.push(installedChunks[chunkId][0]), installedChunks[chunkId] = 0;
        for (moduleId in moreModules) Object.prototype.hasOwnProperty.call(moreModules, moduleId) && (modules[moduleId] = moreModules[moduleId]);
        for (parentJsonpFunction && parentJsonpFunction(chunkIds, moreModules, executeModules); resolves.length; ) resolves.shift()();
        if (executeModules) for (i = 0; i < executeModules.length; i++) result = __webpack_require__(__webpack_require__.s = executeModules[i]);
        return result;
    };
    var installedModules = {}, installedChunks = {
        1: 0
    };
    function __webpack_require__(moduleId) {
        if (installedModules[moduleId]) return installedModules[moduleId].exports;
        var module = installedModules[moduleId] = {
            i: moduleId,
            l: !1,
            exports: {}
        };
        return modules[moduleId].call(module.exports, module, module.exports, __webpack_require__), 
        module.l = !0, module.exports;
    }
    __webpack_require__.m = modules, __webpack_require__.c = installedModules, __webpack_require__.d = function(exports, name, getter) {
        __webpack_require__.o(exports, name) || Object.defineProperty(exports, name, {
            configurable: !1,
            enumerable: !0,
            get: getter
        });
    }, __webpack_require__.n = function(module) {
        var getter = module && module.__esModule ? function getDefault() {
            return module.default;
        } : function getModuleExports() {
            return module;
        };
        return __webpack_require__.d(getter, \\"a\\", getter), getter;
    }, __webpack_require__.o = function(object, property) {
        return Object.prototype.hasOwnProperty.call(object, property);
    }, __webpack_require__.p = \\"\\", __webpack_require__.oe = function(err) {
        throw console.error(err), err;
    };
}([]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`keep_fnames\` option: warnings 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`nameCache\` option: errors 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`nameCache\` option: main.js 1`] = `
"webpackJsonp([ 0 ], [ function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    Object.defineProperty(__webpack_exports__, \\"__esModule\\", {
        value: !0
    });
    var __WEBPACK_IMPORTED_MODULE_0__dep__ = __webpack_require__(1);
    __webpack_exports__.default = function() {
        const b = __WEBPACK_IMPORTED_MODULE_0__dep__.b, baz = \`baz\${Math.random()}\`;
        return () => ({
            a: b + __WEBPACK_IMPORTED_MODULE_0__dep__.a + baz,
            b: b,
            baz: baz
        });
    };
}, function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    __webpack_exports__.a = \\"bar\\", __webpack_exports__.b = \\"foo\\";
} ], [ 0 ]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`nameCache\` option: manifest.js 1`] = `
"!function(modules) {
    var parentJsonpFunction = window.webpackJsonp;
    window.webpackJsonp = function(chunkIds, moreModules, executeModules) {
        for (var moduleId, chunkId, result, i = 0, resolves = []; i < chunkIds.length; i++) chunkId = chunkIds[i], 
        installedChunks[chunkId] && resolves.push(installedChunks[chunkId][0]), installedChunks[chunkId] = 0;
        for (moduleId in moreModules) Object.prototype.hasOwnProperty.call(moreModules, moduleId) && (modules[moduleId] = moreModules[moduleId]);
        for (parentJsonpFunction && parentJsonpFunction(chunkIds, moreModules, executeModules); resolves.length; ) resolves.shift()();
        if (executeModules) for (i = 0; i < executeModules.length; i++) result = __webpack_require__(__webpack_require__.s = executeModules[i]);
        return result;
    };
    var installedModules = {}, installedChunks = {
        1: 0
    };
    function __webpack_require__(moduleId) {
        if (installedModules[moduleId]) return installedModules[moduleId].exports;
        var module = installedModules[moduleId] = {
            i: moduleId,
            l: !1,
            exports: {}
        };
        return modules[moduleId].call(module.exports, module, module.exports, __webpack_require__), 
        module.l = !0, module.exports;
    }
    __webpack_require__.m = modules, __webpack_require__.c = installedModules, __webpack_require__.d = function(exports, name, getter) {
        __webpack_require__.o(exports, name) || Object.defineProperty(exports, name, {
            configurable: !1,
            enumerable: !0,
            get: getter
        });
    }, __webpack_require__.n = function(module) {
        var getter = module && module.__esModule ? function() {
            return module.default;
        } : function() {
            return module;
        };
        return __webpack_require__.d(getter, \\"a\\", getter), getter;
    }, __webpack_require__.o = function(object, property) {
        return Object.prototype.hasOwnProperty.call(object, property);
    }, __webpack_require__.p = \\"\\", __webpack_require__.oe = function(err) {
        throw console.error(err), err;
    };
}([]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`nameCache\` option: warnings 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`safari10\` option: errors 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`safari10\` option: main.js 1`] = `
"webpackJsonp([ 0 ], [ function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    Object.defineProperty(__webpack_exports__, \\"__esModule\\", {
        value: !0
    });
    var __WEBPACK_IMPORTED_MODULE_0__dep__ = __webpack_require__(1);
    __webpack_exports__.default = function() {
        const b = __WEBPACK_IMPORTED_MODULE_0__dep__.b, baz = \`baz\${Math.random()}\`;
        return () => ({
            a: b + __WEBPACK_IMPORTED_MODULE_0__dep__.a + baz,
            b: b,
            baz: baz
        });
    };
}, function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    __webpack_exports__.a = \\"bar\\", __webpack_exports__.b = \\"foo\\";
} ], [ 0 ]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`safari10\` option: manifest.js 1`] = `
"!function(modules) {
    var parentJsonpFunction = window.webpackJsonp;
    window.webpackJsonp = function(chunkIds, moreModules, executeModules) {
        for (var moduleId, chunkId, result, i = 0, resolves = []; i < chunkIds.length; i++) chunkId = chunkIds[i], 
        installedChunks[chunkId] && resolves.push(installedChunks[chunkId][0]), installedChunks[chunkId] = 0;
        for (moduleId in moreModules) Object.prototype.hasOwnProperty.call(moreModules, moduleId) && (modules[moduleId] = moreModules[moduleId]);
        for (parentJsonpFunction && parentJsonpFunction(chunkIds, moreModules, executeModules); resolves.length; ) resolves.shift()();
        if (executeModules) for (i = 0; i < executeModules.length; i++) result = __webpack_require__(__webpack_require__.s = executeModules[i]);
        return result;
    };
    var installedModules = {}, installedChunks = {
        1: 0
    };
    function __webpack_require__(moduleId) {
        if (installedModules[moduleId]) return installedModules[moduleId].exports;
        var module = installedModules[moduleId] = {
            i: moduleId,
            l: !1,
            exports: {}
        };
        return modules[moduleId].call(module.exports, module, module.exports, __webpack_require__), 
        module.l = !0, module.exports;
    }
    __webpack_require__.m = modules, __webpack_require__.c = installedModules, __webpack_require__.d = function(exports, name, getter) {
        __webpack_require__.o(exports, name) || Object.defineProperty(exports, name, {
            configurable: !1,
            enumerable: !0,
            get: getter
        });
    }, __webpack_require__.n = function(module) {
        var getter = module && module.__esModule ? function() {
            return module.default;
        } : function() {
            return module;
        };
        return __webpack_require__.d(getter, \\"a\\", getter), getter;
    }, __webpack_require__.o = function(object, property) {
        return Object.prototype.hasOwnProperty.call(object, property);
    }, __webpack_require__.p = \\"\\", __webpack_require__.oe = function(err) {
        throw console.error(err), err;
    };
}([]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`safari10\` option: warnings 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`toplevel\` option: errors 1`] = `Array []`;

exports[`when applied with uglify-es options matches snapshot for \`toplevel\` option: main.js 1`] = `
"webpackJsonp([ 0 ], [ function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    Object.defineProperty(__webpack_exports__, \\"__esModule\\", {
        value: !0
    });
    var __WEBPACK_IMPORTED_MODULE_0__dep__ = __webpack_require__(1);
    __webpack_exports__.default = function() {
        const b = __WEBPACK_IMPORTED_MODULE_0__dep__.b, baz = \`baz\${Math.random()}\`;
        return () => ({
            a: b + __WEBPACK_IMPORTED_MODULE_0__dep__.a + baz,
            b: b,
            baz: baz
        });
    };
}, function(module, __webpack_exports__, __webpack_require__) {
    \\"use strict\\";
    __webpack_exports__.a = \\"bar\\", __webpack_exports__.b = \\"foo\\";
} ], [ 0 ]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`toplevel\` option: manifest.js 1`] = `
"!function(modules) {
    var parentJsonpFunction = window.webpackJsonp;
    window.webpackJsonp = function(chunkIds, moreModules, executeModules) {
        for (var moduleId, chunkId, result, i = 0, resolves = []; i < chunkIds.length; i++) chunkId = chunkIds[i], 
        installedChunks[chunkId] && resolves.push(installedChunks[chunkId][0]), installedChunks[chunkId] = 0;
        for (moduleId in moreModules) Object.prototype.hasOwnProperty.call(moreModules, moduleId) && (modules[moduleId] = moreModules[moduleId]);
        for (parentJsonpFunction && parentJsonpFunction(chunkIds, moreModules, executeModules); resolves.length; ) resolves.shift()();
        if (executeModules) for (i = 0; i < executeModules.length; i++) result = __webpack_require__(__webpack_require__.s = executeModules[i]);
        return result;
    };
    var installedModules = {}, installedChunks = {
        1: 0
    };
    function __webpack_require__(moduleId) {
        if (installedModules[moduleId]) return installedModules[moduleId].exports;
        var module = installedModules[moduleId] = {
            i: moduleId,
            l: !1,
            exports: {}
        };
        return modules[moduleId].call(module.exports, module, module.exports, __webpack_require__), 
        module.l = !0, module.exports;
    }
    __webpack_require__.m = modules, __webpack_require__.c = installedModules, __webpack_require__.d = function(exports, name, getter) {
        __webpack_require__.o(exports, name) || Object.defineProperty(exports, name, {
            configurable: !1,
            enumerable: !0,
            get: getter
        });
    }, __webpack_require__.n = function(module) {
        var getter = module && module.__esModule ? function() {
            return module.default;
        } : function() {
            return module;
        };
        return __webpack_require__.d(getter, \\"a\\", getter), getter;
    }, __webpack_require__.o = function(object, property) {
        return Object.prototype.hasOwnProperty.call(object, property);
    }, __webpack_require__.p = \\"\\", __webpack_require__.oe = function(err) {
        throw console.error(err), err;
    };
}([]);"
`;

exports[`when applied with uglify-es options matches snapshot for \`toplevel\` option: warnings 1`] = `Array []`;
